Libraries/pbxbuild/Sources/Tool/ResolverCommon.cpp

```diff
@@ -0,0 +1,36 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include <pbxbuild/Tool/ResolverCommon.h>
+#include <process/DefaultContext.h>
+
+namespace Tool = pbxbuild::Tool;
+
+std::unordered_map<std::string, std::string> Tool::ResolverCommon::
+extendedEnvironmentVariables(
+    const std::vector<std::string> &executablePaths,
+    const std::unordered_map<std::string, std::string> &environmentVariables)
+{
+    if (environmentVariables.find("PATH") == environmentVariables.end()) {
+        std::string pathEnvironmentVariable;
+        for (auto path : executablePaths) {
+            pathEnvironmentVariable += path + ":";
+        }
+        auto processPathEnvironmentVariable = process::DefaultContext().environmentVariable("PATH");
```

Comment:
This is just as bad as `getenv()`: it's relying on external process state. To make it clear what state this function uses and to make it testable, it'd need a `process::Context const *` parameter passed in from above.

However, that would be a lot of work to thread through the layers into `pbxbuild::Tool`, and would be a bit unfortunate to require anyway. Luckily, there's an alternative: environment variables are populated in the build settings! So you can just get `PATH` from `environment.resolve("PATH")` on the `pbxsetting::Environment` already available.